### PR TITLE
Add mini winni

### PR DESCRIPTION
**Pull request by matthewdias**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

### CODE REVIEW

File: keyboards/mini_winni/mini_winni.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "quantum.h"
+
+#define XXX KC_NO
+
```
Comment (zvecr) [inline keyboards/mini_winni/mini_winni.h:23]:
```suggestion

```

File: keyboards/mini_winni/info.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "keyboard_name": "Mini Winni",
+    "maintainer": "Matthew Dias",
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/mini_winni/info.json:5]:
```suggestion
        "LAYOUT_all": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/mini_winni/config.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+/* key matrix pins */
+#define DIRECT_PINS { { F4, F5, B4, D7 }, { F6, F7, B6, B5 } }
+#define UNUSED_PINS
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+#define RGB_DI_PIN E6
+#define RGBLED_NUM 4
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/mini_winni/config.h:44]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mini_winni/rules.mk [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = yes	    # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/mini_winni/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/mini_winni/rules.mk [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+MOUSEKEY_ENABLE = yes	    # Mouse keys
+EXTRAKEY_ENABLE = yes	    # Audio control and System control
+CONSOLE_ENABLE = no	        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/mini_winni/rules.mk:17]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — CHANGES_REQUESTED** [review]
This needs a readme

File: keyboards/winkeys/mini_winni/rules.mk [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes	    # Mouse keys
+EXTRAKEY_ENABLE = yes	    # Audio control and System control
+CONSOLE_ENABLE = no	        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes		    # USB Nkey Rollover
+BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/winkeys/mini_winni/rules.mk:18]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no	        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**